components/console/src/core/infrastructure/midaz/repositories/midaz-operation-routes-repository.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  OperationRoutesEntity,
+  OperationRoutesSearchEntity
+} from '@/core/domain/entities/operation-routes-entity'
+import { OperationRoutesRepository } from '@/core/domain/repositories/operation-routes-repository'
+import { injectable, inject } from 'inversify'
+import { PaginationEntity } from '@/core/domain/entities/pagination-entity'
+import { MidazHttpService } from '../services/midaz-http-service'
+import { MidazOperationRoutesDto } from '../dto/midaz-operation-routes.dto'
+import { MidazPaginationDto } from '../dto/midaz-pagination-dto'
+import { MidazOperationRoutesMapper } from '../mappers/midaz-operation-routes-mapper'
+import { createQueryString } from '@/lib/search'
+import { MidazApiException } from '../exceptions/midaz-exceptions'
+
+@injectable()
+export class MidazOperationRoutesRepository implements OperationRoutesRepository {
+  private baseUrl: string = process.env.MIDAZ_TRANSACTION_BASE_PATH as string
+
+  constructor(
+    @inject(MidazHttpService)
+    private readonly httpService: MidazHttpService
+  ) {}
+
+  async create(
+    organizationId: string,
+    ledgerId: string,
+    operationRoute: OperationRoutesEntity
+  ): Promise<OperationRoutesEntity> {
+    const dto = MidazOperationRoutesMapper.toCreateDto(operationRoute)
+
+    const response = await this.httpService.post<MidazOperationRoutesDto>(
+      `${this.baseUrl}/organizations/${organizationId}/ledgers/${ledgerId}/operation-routes`,
+      {
+        body: JSON.stringify(dto)
+      }
+    )
+    return MidazOperationRoutesMapper.toEntity(response)
+  }
+
+  async fetchAll(
+    organizationId: string,
+    ledgerId: string,
+    query?: OperationRoutesSearchEntity
+  ): Promise<PaginationEntity<OperationRoutesEntity>> {
+    const { id, page = 1, limit = 10 } = query ?? {}
+
+    const queryParams = createQueryString({
+      id,
+      page: page.toString(),
+      limit: limit.toString()
+    })
+
+    console.log('queryParams', queryParams)
```

Comment:
Remove the console.log.
You can also call the createQueryString directly into the URL interpolation.